backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DunningBL.java

```diff
@@ -247,6 +252,63 @@ public void processCandidates(final IDunningContext context, final IDunningCandi
 		}
 
 		dunningProducer.finish();
+
+		sendMassNotifications(dunningProducer.getCreatedDunningDocIds(), Language.getBaseLanguage());
+	}
+
+	private void sendMassNotifications(@NonNull final Collection<DunningDocId> createdDunningDocIds,
+			@NonNull final Language language)
+	{
+		dunningDAO.getByIdsInTrx(createdDunningDocIds)
+				.stream()
+				.collect(Collectors.groupingBy(doc -> docTypeDAO.getById(DocTypeId.ofRepoId(doc.getC_DocType_ID())).getMassGenerateNotification()))
```

Comment:
tomatoes tomatoes, chika boom, 

> docTypeDAO.getById(DocTypeId.ofRepoId(doc.getC_DocType_ID())).getMassGenerateNotification()))

i suggest extracting this to a static private method, to have something readable

Reply:
extracted